javascript/hangman.js

```diff
@@ -1,35 +1,60 @@
 class Hangman {
   constructor(words) {
     this.words = words;
-    // ... your code goes here
+    this.secretWord = this.pickWord();
+    this.letters = [];
+    this.guessedLetters = "";
+    this.errorsLeft = 10;
   }
 
   pickWord() {
-    // ... your code goes here
+    return this.words[Math.floor(Math.random() * this.words.length)];
   }
 
-  checkIfLetter(keyCode) {
-    // ... your code goes here
+  checkIfLetter(key) {
+    if(key >= 65 && key <= 90) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
   checkClickedLetters(letter) {
-    // ... your code goes here
+    if(!this.letters.includes(letter)) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
   addCorrectLetter(letter) {
-    // ... your code goes here
+    this.guessedLetters += letter;
+    this.checkWinner();
   }
 
   addWrongLetter(letter) {
-    // ... your code goes here
+    this.errorsLeft--;
+    this.letters.push(letter);
   }
 
   checkGameOver() {
-    // ... your code goes here
+    if(this.errorsLeft == 0) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
```

Comment:
Idem que mas abajo, la funcion podria retornar simplemente `return !this.errorsLeft`